go/private/actions/archive.bzl

```
@@ -64,16 +64,14 @@ def emit_archive(go, source = None, _recompile_suffix = "", recompile_internal_d
     # nogo is a FilesToRunProvider and some targets don't have it, some have it but no executable.
     if nogo != None and nogo.executable != None:
         out_facts = go.declare_file(go, name = source.name, ext = pre_ext + ".facts")
-        out_nogo_log = go.declare_file(go, name = source.name, ext = pre_ext + ".nogo.log")
-        out_nogo_fix = go.declare_file(go, name = source.name, ext = pre_ext + ".nogo.patch")
+        out_nogo = go.declare_directory(go, name = source.name, ext = pre_ext + "_nogo")
```

```suggestion
        out_diagnostics = go.declare_directory(go, name = source.name, ext = pre_ext + "_diagnostics")
```

Maybe calling it "diagnostics" because both the log and fixes are from the nogo diagnostics?